app/assets/javascripts/initializers/initializeArticleReactions.js

```diff
@@ -159,20 +159,26 @@ function initializeArticleReactions() {
   setCollectionFunctionality();
   setTimeout(() => {
     var articleId;
-    if (document.getElementById('article-body')) {
-      articleId = document.getElementById('article-body').dataset.articleId;
-      if (document.getElementById('article-reaction-actions')) {
-        requestReactionCounts(articleId);
-      }
-    }
     var reactionButts = document.getElementsByClassName(
       'article-reaction-butt',
     );
-    for (var i = 0; i < reactionButts.length; i += 1) {
-      reactionButts[i].onclick = function addReactionOnClick(e) {
-        reactToArticle(articleId, this.dataset.category);
-      };
+
+    // we wait for the article to appear,
+    // we also check that reaction buttons are there as draft articles don't have them
+    if (document.getElementById('article-body') && reactionButts.length > 0) {
+      articleId = document.getElementById('article-body').dataset.articleId;
+
+      if (reactionButts.length > 0) {
```

Comment:
Isn't this condition already part of the surrounding `if`?

Reply:
done :) thanks!